include/boost/json/error.hpp

```diff
@@ -101,6 +101,16 @@ enum class error
 
     /// past-the-end index is not supported
     past_the_end,
+
+    //
+    // Conversion errors
+    //
+
+    /// JSON bool was expected during conversion
+    bool_expected,
+
+    /// JSON string was expected during conversion
+    string_expected,
```

Comment:
when you add errors these need to be added to error.ipp and error.cpp (tests) as well

Reply:
Don't review this, it's just an outline